assets/src/block-editor/components/amp-preview.js

```diff
@@ -0,0 +1,318 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { Component, createRef, renderToString } from '@wordpress/element';
+import { Icon, IconButton } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { withSelect, withDispatch } from '@wordpress/data';
+import { DotTip } from '@wordpress/nux';
+import { compose } from '@wordpress/compose';
+import { addQueryArgs } from '@wordpress/url';
+
+/**
+ * Internal dependencies
+ */
+import ampBlackIcon from '../../../images/amp-black-icon.svg';
+import ampFilledIcon from '../../../images/amp-icon.svg';
+import { isAMPEnabled } from '../helpers';
+import { POST_PREVIEW_CLASS } from '../constants';
+
+/**
+ * Writes the message and graphic in the new preview window that was opened.
+ *
+ * Forked from the Core component <PostPreviewButton>.
+ *
+ * @see https://github.com/WordPress/gutenberg/blob/95e769df1f82f6b0ef587d81af65dd2f48cd1c38/packages/editor/src/components/post-preview-button/index.js#L17
+ * @param {Object} targetDocument The target document.
```

Comment:
This can be more explicit:

```suggestion
 * @param {Document} targetDocument The target document.
```

Reply:
Good point